### ISSUE TITLE
recent update leads to failure to load jinx module

### ISSUE THREAD

**Issue by ericsfraga** [minad/jinx#111]
I updated to jinx-20231015.1701 and, since then, I have been unable to have jinx start up. I get

`jinx--load-module: Module initialization failed: "/home/ucecesf/.emacs.d/elpa,v30/jinx-20231015.1701/jinx-mod.so", 3`

I've tried uninstalling and re-installing but to no effect. This is on ubuntu with Emacs version 30 from git (as of a few weeks ago).

**Comment by ericsfraga** [comment]
Sorry but where does that leave me? I cannot upgrade the OS so does that mean I have to stop using jinx? Pity if so.

**Comment by minad** [comment]
I suggest to pin Jinx to the stable version 0.9 for the time being. Which Ubuntu version do you use? When do you expect an update? For me Debian is still the gold standard with respect to stability and if something is more dated than Debian stable, for now I consider it acceptable to drop support. Note that Jinx is still quite young ("in development"). It also looks as if Enchant development recently picked up steam, see https://github.com/AbiWord/enchant/blob/master/NEWS. There are even new changes/fixes in Enchant, discovered and contributed by Jinx users.

**Comment by ericsfraga** [comment]
Okay, thank you. I will try pinning to the stable version. I am using Ubuntu 20.04 LTS and do not expect to upgrade in quite some time, hence the LTS version. This is a working computer (36 core) used by my team (but is also my desktop) and we minimise upgrades (except for security).

I do use Debian on most of my other systems but tend to stick to the stable release there (which means I upgrade a bit more often).

**Comment by ericsfraga** [comment]
Yeah, it can be challenging to keep supporting old systems... and I fully understand not wishing to! But for production systems, like mine, stability wins over keeping up with upgrades. Luckily, with Emacs, I have options so no worries. Thanks again.

**Comment by minad** [comment]
On 10/16/23 19:34, Tomasz Hołubowicz wrote:
> For anyone that uses Ubuntu 20.04 and still wants to use |jinx| below
> snippet should be enough to compile it from source (let me know if I
> missed something as some dependencies could have been already installed
> on my machine):
> 
> wget https://github.com/AbiWord/enchant/releases/download/v2.6.1/enchant-2.6.1.tar.gz &&
>     tar -xf enchant-2.6.1.tar.gz &&
>     cd enchant-2.6.1 &&
>     sudo apt install groff libglib2.0-dev libhunspell-dev &&
>     ./configure --with-hunspell --with-hunspell-dir=/usr/share/hunspell --enable-relocatable &&
>     make && sudo make install && sudo ldconfig &&
>     cd .. && rm -rf enchant*
> 
> (|--with-hunspell| can be replaced with other spelling provider like
> |aspell| or |nuspell|. For other options run |./configure --help|)

Thanks. Note that this isn't be needed if you can live with the message
about the Enchant version.

**Comment by alternateved** [comment]
Ah, I didn't realize that it was working again with outdated version. Great, thanks!
